### PR TITLE
Updated the discovery api and storage pool spec

### PR DESCRIPTION
**Pull request by kushagra-mahajan**
- [X] I have read the [CONTRIBUTION GUIDE][contributing]

- [X] This is a bug

- [X] This is a feature request

- [X] This is a syntax/style fix

- [X] This is an improvement to code quality

---
**Please describe your changes below:**
(Note: A clear description of changes is required to approve any pull requests)

_"Changed the implementation for verification because..."_

[contributing]: http://dellemc-symphony.readthedocs.io/en/latest/contributingtosymphony.html

### CODE REVIEW

File: src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOProtectionDomainDataRestRep.jsd [hunk 1]
```diff
@@ -14,6 +13,12 @@
     },
     "protectionDomainState" : {
       "type" : "string"
+    },
+    "rfcacheEnabled" : {
```
Comment (silvam11) [inline src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOProtectionDomainDataRestRep.jsd]: Should it be rfCache...

File: src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOStoragePoolDataRestRep.jsd [hunk 2]
```diff
@@ -20,6 +19,18 @@
...
+    "useRfcache" : {
+      "type" : "boolean"
+    },
+    "useRmcache" : {
+      "type" : "boolean"
+    },
+    "zeroPaddingEnabled" : {
+      "type" : "boolean"
+    },
+    "rmcacheWriteHandlingMode" : {
```
Comment (silvam11) [inline src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOStoragePoolDataRestRep.jsd]: Should it be rmCache ... ?

File: src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOProtectionDomainDataRestRep.jsd (hunk 1 above)
Comment (kushagra-mahajan) [reply src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOProtectionDomainDataRestRep.jsd]: Its fine we followed the REST API, same variable names

File: src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOStoragePoolDataRestRep.jsd (hunk 2 above)
Comment (kushagra-mahajan) [reply src/main/resources/storage-capabilities-api/schema/json/includes/ScaleIOStoragePoolDataRestRep.jsd]: Its fine we followed the REST API, same variable names

**Pull request merged**